src/standard_binary_tree.rs

```diff
@@ -76,12 +76,15 @@ impl StandardMerkleTree {
     /// Constructs a [`StandardMerkleTree`] from a slice of dynamic Solidity values.
     pub fn of(values: &[DynSolValue]) -> Self {
         // Hash each value and associate it with its index and leaf hash.
-        let hashed_values: Vec<(&DynSolValue, usize, B256)> = values
+        let mut hashed_values: Vec<(&DynSolValue, usize, B256)> = values
             .iter()
             .enumerate()
             .map(|(i, value)| (value, i, standard_leaf_hash(value)))
             .collect();
 
+        // Sort the hashed values by their hash.
```

Comment:
imo this should be optional e.g oz merkle tree implementation passing sort flag as option (https://github.com/OpenZeppelin/merkle-tree/blob/master/src/options.ts#L2).

e.g
```rust
pub fn of(values: &[DynSolValue], sortLeaves: Option<bool>) -> Self
```